src/term/check/unbound_vars.rs

```diff
@@ -25,8 +25,11 @@ impl Term {
 
     // Check global vars
     for (nam, (declared, used)) in globals.into_iter() {
-      if used && !declared {
-        return Err(format!("Unbound unscoped variable '${nam}'"));
+      match (used, declared) {
+        (1, 1) => {}
+        (0, _) => return Err(format!("Missing unscoped variable use for 'λ${nam}'")),
+        (_, 0) => return Err(format!("Unbound unscoped variable '${nam}'")),
+        (_, _) => return Err(format!("Unscoped variable '${nam}' used more than once")),
```

Comment:
A bit confusing that you're mixing 1 declaration multiple use and multiple declarations.